core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java

```diff
@@ -0,0 +1,346 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.exception.AccessControlException;
+import alluxio.exception.ExceptionMessage;
+import alluxio.exception.InvalidPathException;
+import alluxio.exception.PreconditionMessage;
+import alluxio.master.file.meta.Inode;
+import alluxio.master.file.meta.InodeTree;
+import alluxio.master.file.meta.LockedInodePath;
+import alluxio.security.authentication.AuthenticatedClientUser;
+import alluxio.security.authorization.Mode;
+import alluxio.util.CommonUtils;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.base.Preconditions;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Default implementation of permission checker.
+ */
+// TODO(peis): Migrate this class to a set of static functions.
+@NotThreadSafe // TODO(jiri): make thread-safe (c.f. ALLUXIO-1664)
+public final class DefaultPermissionChecker implements PermissionChecker{
+  /** The file system inode structure. */
+  private final InodeTree mInodeTree;
+
+  /** Whether the permission check is enabled. */
+  private final boolean mPermissionCheckEnabled;
+
+  /** The super group of Alluxio file system. All users in this group have super permission. */
+  private final String mFileSystemSuperGroup;
+
+  private final InodePermissionChecker mInodePermissionChecker;
+
+  /**
+   * Constructs a {@link PermissionChecker} instance for Alluxio file system.
+   *
+   * @param inodeTree inode tree of the file system master
+   */
+  public DefaultPermissionChecker(InodeTree inodeTree) {
+    mInodeTree = Preconditions.checkNotNull(inodeTree, "inodeTree");
+    mPermissionCheckEnabled =
+        Configuration.getBoolean(PropertyKey.SECURITY_AUTHORIZATION_PERMISSION_ENABLED);
+    mFileSystemSuperGroup =
+        Configuration.get(PropertyKey.SECURITY_AUTHORIZATION_PERMISSION_SUPERGROUP);
+    mInodePermissionChecker = InodePermissionChecker.create();
+  }
+
+  /**
+   * Checks whether a user has permission to perform a specific action on the parent of the given
```

Comment:
NIT: does the interface already have these comments? If so, you can remove these.

Reply:
Done.